internal/codespaces/codespaces.go

```diff
@@ -43,9 +60,43 @@ func (e *TimeoutError) Error() string {
 	return e.message
 }
 
-// ConnectToLiveshare waits for a Codespace to become running,
-// and connects to it using a Live Share session.
+// GetCodespaceConnection waits until a codespace is able
+// to be connected to and initializes a connection to it.
+func GetCodespaceConnection(ctx context.Context, progress progressIndicator, apiClient apiClient, codespace *api.Codespace, httpClient *http.Client) (*connection.CodespaceConnection, error) {
```

Comment:
This feels real odd passing in both an `apiClient` and a `httpClient`. It would be nice to keep the `apiClient` abstraction throughout the codespaces commands for consistency. Considering that `apiClient` actually has an `*http.Client` in its implementation, perhaps all we really need to do is create a method to expose it for use with the tunnel package.

Reply:
I didn't realize the `apiClient` had an HTTP client already attached. I've added a `GetHttpClient` function to be able to access that client rather than passing it all the way through.